deps/rabbit/src/rabbit_variable_queue.erl

```diff
@@ -1033,6 +1111,184 @@ wait_for_msg_store_credit() ->
+    V1Index0 = rabbit_queue_index:init_for_conversion(QueueName, MsgIdxOnDiskFun, MsgAndIdxOnDiskFun),
+    {LoSeqId, HiSeqId, _} = rabbit_classic_queue_index_v2:bounds(V2Index),
+    V1Index = convert_from_v2_to_v1_loop(QueueName, V1Index0, V2Index, V2Store, LoSeqId, HiSeqId),
+    %% We have already closed the v2 index/store FDs when deleting the files.
+    State#vqstate{ index_mod   = rabbit_queue_index,
+                   index_state = V1Index,
+                   store_state = undefined }.
+
+convert_from_v2_to_v1_in_memory(State0 = #vqstate{ q1 = Q1b,
+                                                   q2 = Q2b,
+                                                   q3 = Q3b,
+                                                   q4 = Q4b }) ->
+    {Q1, State1} = convert_from_v2_to_v1_queue(Q1b, State0),
+    {Q2, State2} = convert_from_v2_to_v1_queue(Q2b, State1),
+    {Q3, State3} = convert_from_v2_to_v1_queue(Q3b, State2),
+    {Q4, State4} = convert_from_v2_to_v1_queue(Q4b, State3),
+    State4#vqstate{ q1 = Q1,
+                    q2 = Q2,
+                    q3 = Q3,
+                    q4 = Q4 }.
+
+%% We fetch the message from the per-queue store if necessary
+%% and mark all messages as delivered to make the v1 index happy.
+convert_from_v2_to_v1_queue(Q, State0) ->
+    List0 = ?QUEUE:to_list(Q),
+    {List, State} = lists:mapfoldl(fun (MsgStatus0, State1 = #vqstate{ store_state = StoreState0 }) ->
+        case MsgStatus0 of
+            #msg_status{ seq_id = SeqId,
+                         msg_location = MsgLocation = {rabbit_classic_queue_store_v2, _, _} } ->
+                {Msg, StoreState} = rabbit_classic_queue_store_v2:read(SeqId, MsgLocation, StoreState0),
+                MsgStatus = MsgStatus0#msg_status{ msg = Msg,
+                                                   msg_location = memory,
+                                                   is_delivered = true,
+                                                   persist_to   = queue_index },
+                State2 = stats(ready0, {MsgStatus0, MsgStatus}, 0, State1),
+                {MsgStatus, State2#vqstate{ store_state = StoreState }};
+            #msg_status{ persist_to = queue_store } ->
+                {MsgStatus0#msg_status{ is_delivered = true,
+                                        persist_to   = queue_index }, State1};
+            _ ->
+                {MsgStatus0#msg_status{ is_delivered = true }, State1}
+        end
+    end, State0, List0),
+    {?QUEUE:from_list(List), State}.
+
+convert_from_v2_to_v1_loop(_, V1Index, _, _, HiSeqId, HiSeqId) ->
+    V1Index;
+convert_from_v2_to_v1_loop(QueueName, V1Index0, V2Index0, V2Store0, LoSeqId, HiSeqId) ->
+    UpSeqId = lists:min([rabbit_classic_queue_index_v2:next_segment_boundary(LoSeqId),
+                         HiSeqId]),
+    {Messages, V2Index1} = rabbit_classic_queue_index_v2:read(LoSeqId, UpSeqId, V2Index0),
+    logger:error("HEREHERE~n~p~n~p", [Messages, V2Index0]),
```

Comment:
Drop this and the other 2 logger:error calls (or commented out logger:error calls).